cloudify_awssdk/common/decorators.py

```diff
@@ -135,6 +135,16 @@ def wrapper_inner(**kwargs):
                     ctx.instance.runtime_properties[
                         'resource_config'] = resource_config
                     ctx.instance.runtime_properties[EXT_RES_ID] = resource_id
+                    _, _, _, operation_name = ctx.operation.name.split('.')
+                    if operation_name not in ['delete', 'create'] and \
+                            ctx.node.properties['verify_external_resource'] \
+                            and not \
+                            kwargs['iface'].resource_id_in_properties():
+                        raise NonRecoverableError(
+                            'Resource type {0} resource_id '
+                            '{1} not found.'.format(
+                                kwargs['resource_type'],
+                                kwargs['iface'].resource_id))
```

Comment:
@EarthmanT Is this a special case for IP or we can apply it on ant other AWS resources ?

Reply:
Yes I want to. I just did not want to do all of the work and testing, without first checking out the solution.